activerecord/lib/active_record/model_schema.rb

```diff
@@ -477,6 +484,7 @@ def load_schema!
         def reload_schema_from_cache
           @arel_table = nil
           @column_names = nil
+          @symbol_column_to_string_name_hash = nil
```

Comment:
@sgrif is there a place I should be testing that this works? I poked around a bit an didn't see a centralized location for testing this behavior. Also, thoughts on the general concept of the PR and the implementation are appreciated if you have time.